sycl/doc/syclcompat/README.md

```diff
@@ -72,6 +72,40 @@ This document presents the public API under the [Features](#features) section,
 and provides a working [Sample code](#sample-code) using this library. Refer to
 those to learn to use the library.
 
+## Versioning
+
+SYCLcompat adopts [semantic versioning](https://semver.org/)
+(`major.minor.patch`) in a manner which aligns with oneAPI releases. Each oneAPI
+product release has an associated SYCLcompat release. When breaking changes have
+been made to SYCLcompat's API since the last release, `major` is incremented,
+otherwise `minor` is incremented. Use of `patch` is described below.
+
+Between release cycles, ongoing updates to SYCLcompat (including possibly
+breaking changes) are merged into DPC++'s
+[`sycl`](https://github.com/intel/llvm/tree/sycl) branch. To enable projects
+pulling SYCLcompat from source to distinguish from the latest release version,
+the `patch` version in the `sycl` branch is incremented immediately following a
+oneAPI release.
+
+If important updates (e.g. bugfixes, breaking changes) are pushed to the `sycl`
```

Comment:
I think any breaking change should increase the major version regardless, or users may get a big surprise. If some other breaking change with the previous version is introduced, we then DON'T bump again. (I.e. as much as 1 major bump per release).